rest/route/user.go

```diff
@@ -852,16 +852,20 @@ func (h *renameUserHandler) Parse(ctx context.Context, r *http.Request) error {
 }
 
 func (h *renameUserHandler) Run(ctx context.Context) gimlet.Responder {
+	githubUID := h.oldUsr.Settings.GithubUser.UID
+	h.oldUsr.Settings.GithubUser.UID = 0
 	// First, upsert the new user. If this doesn't work, there's no reason to continue.
 	newUsr, err := user.UpsertOneFromExisting(h.oldUsr, h.newEmail)
 	if err != nil {
 		return gimlet.NewJSONInternalErrorResponse(err)
 	}
 
-	// Next, remove the old user, consolidate patches, update the host, and handle the error.
 	catcher := grip.NewBasicCatcher()
 	catcher.Add(user.ClearUser(h.oldUsr.Id))
-	catcher.Add(h.oldUsr.UpdateAPIKey("")) // Unset API key to avoid duplicates.
+	// Update the new user's github UID, now that we've cleared out the old user.
```

Comment:
Rather than state what this code is doing (which looks pretty clear just from reading), can we say that we specifically have to clear and then set the GitHub username in this order because the DB has a uniqueness constraint that prevents users from simultaneously having the same GitHub UID? Would be more useful to understand why we're doing something odd, rather than just saying what we're doing.